Syncing failure with rnaseq + viralrecon

Syncing of `rnaseq` fails


[rnaseq_sync_log.zip](https://github.com/nf-core/tools/files/6164098/rnaseq_sync_log.zip)


```
Run git config --global user.email "core@nf-co.re"

                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

    nf-core/tools version 1.13



INFO     Pipeline directory:                                          sync.py:84
         /home/runner/work/tools/tools/nf-core/rnaseq                           
INFO     Using branch 'dev' to fetch workflow variables               sync.py:86
INFO     Will attempt to automatically create a pull request          sync.py:88
INFO     Original pipeline repository branch is 'dev'                sync.py:140
╭───────────────────── Traceback (most recent call last) ──────────────────────╮
│ /opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/nf_core/ut │
│ ils.py:267 in nextflow_cmd                                                   │
│                                                                              │
│   264 def nextflow_cmd(cmd):                                                 │
│   265 │   """Run a Nextflow command and capture the output. Handle errors ni │
│   266 │   try:                                                               │
│ ❱ 267 │   │   nf_proc = subprocess.run(shlex.split(cmd), stdout=subprocess.P │
│       stderr=subprocess.PIPE, check=True)                                    │
│   268 │   │   return nf_proc.stdout                                          │
│   269 │   except OSError as e:                                               │
│   270 │   │   if e.errno == errno.ENOENT:                                    │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/subprocess.py:516 in run │
│                                                                              │
│    513 │   │   │   raise                                                     │
│    514 │   │   retcode = process.poll()                                      │
│    515 │   │   if check and retcode:                                         │
│ ❱  516 │   │   │   raise CalledProcessError(retcode, process.args,           │
│    517 │   │   │   │   │   │   │   │   │    output=stdout, stderr=stderr)    │
│    518 │   return CompletedProcess(process.args, retcode, stdout, stderr)    │
╰──────────────────────────────────────────────────────────────────────────────╯
CalledProcessError: Command '['nextflow', 'config', '-flat', 
'/home/runner/work/tools/tools/nf-core/rnaseq']' returned non-zero exit status 
1.

During handling of the above exception, another exception occurred:

╭───────────────────── Traceback (most recent call last) ──────────────────────╮
│ /opt/hostedtoolcache/Python/3.8.8/x64/bin/nf-core:8 in <module>              │
│                                                                              │
│   5 from nf_core.__main__ import run_nf_core                                 │
│   6 if __name__ == '__main__':                                               │
│   7 │   sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])     │
│ ❱ 8 │   sys.exit(run_nf_core())                                              │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/nf_core/__ │
│ main__.py:57 in run_nf_core                                                  │
│                                                                              │
│    54 │   stderr.print("\n\n")                                               │
│    55 │                                                                      │
│    56 │   # Lanch the click cli                                              │
│ ❱  57 │   nf_core_cli()                                                      │
│    58                                                                        │
│    59                                                                        │
│    60 # Customise the order of subcommands for --help                        │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/click/core │
│ .py:829 in __call__                                                          │
│                                                                              │
│    826 │                                                                     │
│    827 │   def __call__(self, *args, **kwargs):                              │
│    828 │   │   """Alias for :meth:`main`."""                                 │
│ ❱  829 │   │   return self.main(*args, **kwargs)                             │
│    830                                                                       │
│    831                                                                       │
│    832 class Command(BaseCommand):                                           │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/click/core │
│ .py:782 in main                                                              │
│                                                                              │
│    779 │   │   try:                                                          │
│    780 │   │   │   try:                                                      │
│    781 │   │   │   │   with self.make_context(prog_name, args, **extra) as c │
│ ❱  782 │   │   │   │   │   rv = self.invoke(ctx)                             │
│    783 │   │   │   │   │   if not standalone_mode:                           │
│    784 │   │   │   │   │   │   return rv                                     │
│    785 │   │   │   │   │   # it's not safe to `ctx.exit(rv)` here!           │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/click/core │
│ .py:1259 in invoke                                                           │
│                                                                              │
│   1256 │   │   │   │   Command.invoke(self, ctx)                             │
│   1257 │   │   │   │   sub_ctx = cmd.make_context(cmd_name, args, parent=ctx │
│   1258 │   │   │   │   with sub_ctx:                                         │
│ ❱ 1259 │   │   │   │   │   return _process_result(sub_ctx.command.invoke(sub │
│   1260 │   │                                                                 │
│   1261 │   │   # In chain mode we create the contexts step by step, but afte │
│   1262 │   │   # base command has been invoked.  Because at that point we do │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/click/core │
│ .py:1066 in invoke                                                           │
│                                                                              │
│   1063 │   │   """                                                           │
│   1064 │   │   _maybe_show_deprecated_notice(self)                           │
│   1065 │   │   if self.callback is not None:                                 │
│ ❱ 1066 │   │   │   return ctx.invoke(self.callback, **ctx.params)            │
│   1067                                                                       │
│   1068                                                                       │
│   1069 class MultiCommand(Command):                                          │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/click/core │
│ .py:610 in invoke                                                            │
│                                                                              │
│    607 │   │   args = args[2:]                                               │
│    608 │   │   with augment_usage_errors(self):                              │
│    609 │   │   │   with ctx:                                                 │
│ ❱  610 │   │   │   │   return callback(*args, **kwargs)                      │
│    611 │                                                                     │
│    612 │   def forward(*args, **kwargs):  # noqa: B902                       │
│    613 │   │   """Similar to :meth:`invoke` but fills in default keyword     │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/nf_core/__ │
│ main__.py:664 in sync                                                        │
│                                                                              │
│   661 │   # Sync the given pipeline dir                                      │
│   662 │   sync_obj = nf_core.sync.PipelineSync(pipeline_dir, from_branch, pu │
│       repository, username)                                                  │
│   663 │   try:                                                               │
│ ❱ 664 │   │   sync_obj.sync()                                                │
│   665 │   except (nf_core.sync.SyncException, nf_core.sync.PullRequestExcept │
│   666 │   │   log.error(e)                                                   │
│   667 │   │   sys.exit(1)                                                    │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/nf_core/sy │
│ nc.py:91 in sync                                                             │
│                                                                              │
│    88 │   │   │   log.info("Will attempt to automatically create a pull requ │
│    89 │   │                                                                  │
│    90 │   │   self.inspect_sync_dir()                                        │
│ ❱  91 │   │   self.get_wf_config()                                           │
│    92 │   │   self.checkout_template_branch()                                │
│    93 │   │   self.delete_template_branch_files()                            │
│    94 │   │   self.make_template_pipeline()                                  │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/nf_core/sy │
│ nc.py:169 in get_wf_config                                                   │
│                                                                              │
│   166 │   │                                                                  │
│   167 │   │   # Fetch workflow variables                                     │
│   168 │   │   log.debug("Fetching workflow config variables")                │
│ ❱ 169 │   │   self.wf_config = nf_core.utils.fetch_wf_config(self.pipeline_d │
│   170 │   │                                                                  │
│   171 │   │   # Check that we have the required variables                    │
│   172 │   │   for rvar in self.required_config_vars:                         │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/nf_core/ut │
│ ils.py:234 in fetch_wf_config                                                │
│                                                                              │
│   231 │   log.debug("No config cache found")                                 │
│   232 │                                                                      │
│   233 │   # Call `nextflow config`                                           │
│ ❱ 234 │   nfconfig_raw = nextflow_cmd(f"nextflow config -flat {wf_path}")    │
│   235 │   for l in nfconfig_raw.splitlines():                                │
│   236 │   │   ul = l.decode("utf-8")                                         │
│   237 │   │   try:                                                           │
│                                                                              │
│ /opt/hostedtoolcache/Python/3.8.8/x64/lib/python3.8/site-packages/nf_core/ut │
│ ils.py:273 in nextflow_cmd                                                   │
│                                                                              │
│   270 │   │   if e.errno == errno.ENOENT:                                    │
│   271 │   │   │   raise AssertionError("It looks like Nextflow is not instal │
│       required for most nf-core functions.")                                 │
│   272 │   except subprocess.CalledProcessError as e:                         │
│ ❱ 273 │   │   raise AssertionError(                                          │
│   274 │   │   │   f"Command '{cmd}' returned non-zero error code '{e.returnc │
│       {e.stderr.decode()}"                                                   │
│   275 │   │   )                                                              │
╰──────────────────────────────────────────────────────────────────────────────╯
AssertionError: Command 'nextflow config -flat 
/home/runner/work/tools/tools/nf-core/rnaseq' returned non-zero error code '1':
[red]> 
Error: Process completed with exit code 1.
```

--- a/.github/RELEASE_CHECKLIST.md
+++ b/.github/RELEASE_CHECKLIST.md
@@ -2,13 +2,14 @@
 
 1. Check issue milestones to see outstanding issues to resolve if possible or transfer to the milestones for the next release e.g. [`v1.9`](https://github.com/nf-core/tools/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.9)
 2. Most importantly, pick an undeniably outstanding [name](http://www.codenamegenerator.com/) for the release where *Prefix* = *Metal* and *Dictionary* = *Animal*.
-3. Create a PR to `dev` to bump the version in  `CHANGELOG.md` and `setup.py`.
-4. Make sure all CI tests are passing!
-5. Create a PR from `dev` to `master`
-6. Make sure all CI tests are passing again (additional tests are run on PRs to `master`)
-7. Request review (2 approvals required)
-8. Merge the PR into `master`
-9. Once CI tests on commit have passed, create a new release copying the `CHANGELOG` for that release into the description section.
+3. Check whether the GitHub Actions workflow scripts need updating of the Nextflow versions
+4. Create a PR to `dev` to bump the version in  `CHANGELOG.md` and `setup.py`.
+5. Make sure all CI tests are passing!
+6. Create a PR from `dev` to `master`
+7. Make sure all CI tests are passing again (additional tests are run on PRs to `master`)
+8. Request review (2 approvals required)
+9. Merge the PR into `master`
+10. Once CI tests on commit have passed, create a new release copying the `CHANGELOG` for that release into the description section.
 
 ## After release
 

--- a/.github/workflows/create-lint-wf.yml
+++ b/.github/workflows/create-lint-wf.yml
@@ -4,6 +4,8 @@ on: [push, pull_request]
 jobs:
   MakeTestWorkflow:
     runs-on: ubuntu-latest
+    env:
+      NXF_VER: 21.03.0-edge
     steps:
       - uses: actions/checkout@v2
         name: Check out source-code repository

--- a/.github/workflows/pytest.yml
+++ b/.github/workflows/pytest.yml
@@ -4,47 +4,48 @@ name: Python tests
 on:
   push:
     paths:
-      - '**.py'
+      - "**.py"
   pull_request:
     paths:
-      - '**.py'
+      - "**.py"
 
 jobs:
   pytest:
-
     runs-on: ubuntu-latest
+    env:
+      NXF_VER: 21.03.0-edge
     strategy:
       matrix:
         python-version: [3.6, 3.7, 3.8]
 
     steps:
-    - uses: actions/checkout@v2
-      name: Check out source-code repository
+      - uses: actions/checkout@v2
+        name: Check out source-code repository
 
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v1
-      with:
-        python-version: ${{ matrix.python-version }}
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
 
-    - name: Install python dependencies
-      run: |
-        python -m pip install --upgrade pip -r requirements-dev.txt
-        pip install -e .
+      - name: Install python dependencies
+        run: |
+          python -m pip install --upgrade pip -r requirements-dev.txt
+          pip install -e .
 
-    - name: Install Nextflow
-      env:
+      - name: Install Nextflow
+        env:
           CAPSULE_LOG: none
-      run: |
-        mkdir /tmp/nextflow
-        cd /tmp/nextflow
-        wget -qO- get.nextflow.io | bash
-        sudo ln -s /tmp/nextflow/nextflow /usr/local/bin/nextflow
+        run: |
+          mkdir /tmp/nextflow
+          cd /tmp/nextflow
+          wget -qO- get.nextflow.io | bash
+          sudo ln -s /tmp/nextflow/nextflow /usr/local/bin/nextflow
 
-    - name: Test with pytest
-      run: python3 -m pytest --color=yes --cov-report=xml --cov-config=.github/.coveragerc --cov=nf_core
+      - name: Test with pytest
+        run: python3 -m pytest --color=yes --cov-report=xml --cov-config=.github/.coveragerc --cov=nf_core
 
-    - uses: codecov/codecov-action@v1
-      name: Upload code coverage report
-      with:
-        if: success()
-        token: ${{ secrets.CODECOV_TOKEN }}
+      - uses: codecov/codecov-action@v1
+        name: Upload code coverage report
+        with:
+          if: success()
+          token: ${{ secrets.CODECOV_TOKEN }}

--- a/.github/workflows/sync.yml
+++ b/.github/workflows/sync.yml
@@ -7,6 +7,8 @@ on:
 jobs:
   get-pipelines:
     runs-on: ubuntu-latest
+    env:
+      NXF_VER: 21.03.0-edge
     outputs:
       matrix: ${{ steps.set-matrix.outputs.matrix }}
     steps:

--- a/nf_core/pipeline-template/.github/workflows/ci.yml
+++ b/nf_core/pipeline-template/.github/workflows/ci.yml
@@ -20,7 +20,7 @@ jobs:
     strategy:
       matrix:
         # Nextflow versions: check pipeline minimum and current latest
-        nxf_ver: ['20.04.0', '']
+        nxf_ver: ['20.04.0', '21.03.0-edge']
     steps:
       - name: Check out pipeline code
         uses: actions/checkout@v2

